chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs

```diff
@@ -278,13 +279,21 @@ pub fn check_transferred_amount_in_reward<U: UtxosView, P: PoSAccountingView>(
         },
     )?;
 
-    let max_allowed_outputs_total =
-        amount_sum!(inputs_total, block_subsidy_at_height.0, total_fees.0)
-            .ok_or_else(|| ConnectTransactionError::RewardAdditionError(block_id))?;
+    let max_allowed_outputs_total = match consensus_data {
+        ConsensusData::None | ConsensusData::PoW(_) => {
+            amount_sum!(inputs_total, block_subsidy_at_height.0, total_fees.0)
+                .ok_or_else(|| ConnectTransactionError::RewardAdditionError(block_id))?
+        }
+        ConsensusData::PoS(_) => {
+            // TODO: it doesn't actually make much sense to check amounts for PoS, because reward doesn't
+            // transfer anything in terms of utxos.
+            inputs_total
```

Comment:
Are there tests for this behavior? Let's make sure this is tested so that it cannot be abused in a way we can't imagine. When you're writing this test, make sure all holes are closed.